src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -752,7 +752,7 @@ internal static string GetStringForPartUri(string escapedPath)
             Uri safeUnescapedUri = new Uri(partUri.GetComponents(UriComponents.Path, UriFormat.SafeUnescaped), UriKind.Relative);
 
             //Get the escaped string for the part name as part names should have only ascii characters
-            return safeUnescapedUri.GetComponents(UriComponents.SerializationInfoString, UriFormat.UriEscaped);
+            return safeUnescapedUri.GetComponents(UriComponents.SerializationInfoString, UriFormat.Unescaped);
```

Comment:
Does the comment above this line need an update now?

Reply:
good catch, yes.